packages/cli/src/commands/version/index.ts

```
@@ -18,7 +18,16 @@ let importantEnd = chalk.red(
   "----------------------------------------------------------------------"
 );
 
-export default async function version(cwd: string, config: Config) {
+export default async function version(
+  cwd: string, 
+  { ignore }: { ignore?: string[] },
+  config: Config) {
+  // Commandline arguments take precedence over config object,
```

I would prefer merging both - it would be a surprise to me that one-off usage (through CLI) would override the one that is committed to the config file.

---

I think merging might be confusing. I will have to remember what is set in the config file when using `--ignore` in the cli in order to understand the result correctly. I'm not confident that I will always remember it correctly, or I will have to keep going back to the config file when using cli, which doesn't feel like a good experience.

Typescript cli flags override options set in `tsconfig.json`. For example, `tsc --lib ES5` will override the `lib` option set in `tsconfig.json`. So I actually think many people will expect the override behavior. WDYT?

---

Maybe we should just disallow mixing those 2 right now so we don't have to make this decision right now? @mitchellhamilton @Noviny what do you think - what would be the best behavior for this?

---

I would say we shouldn't allow it as a command line argument precisely because it raises questions like this and regardless of what we choose, people might want different behavior + having command line arguments makes it harder to reason about what Changesets/these kinds of tools in general do in a repo.

---

The CLI flag is supposed to help this - https://github.com/atlassian/changesets/issues/286, for which having to update the config file, doing the publish and reverting changes to the config file seems like a bad DX. 

I'm totally fine with disallowing mixing those 2 (config option and CLI flag) ever, such decision shouldn't affect any expectations because we won't provide any default strategy for handling this - instead, we will just fail fast and hard.

---

> we will just fail fast and hard.

This sounds good to me.

---

Updated to throw when `--ignore` and ignore in the config are used at the same time.